src/pages/api/fetch-datasets.ts

```diff
@@ -0,0 +1,14 @@
+import type {NextApiRequest, NextApiResponse} from 'next';
+import {DatasetFetcher} from '@/lib/dataset-fetcher';
+
+const datasetFetcher = new DatasetFetcher({
```

Comment:
An instance of the dataset fetcher is also created in `src/app/[locale]/page.tsx`. Is there a way to make just one instance and re-use that one?

Reply:
I don't think so. I can move the creation of the instance to a generic file (maybe in the fetcher file?). But then still two instances will be created. I don't think I can use one instance in both the nextjs API and nextjs app.

Reply:
About a generic file: why wouldn't this work? If we import that file in both `src/app/[locale]/page.tsx` and `src/pages/api/datasets.ts`, we would have one instance of the dataset fetcher? Or am I overlooking something?